### DISCUSSION TITLE
[Help!] Cluster using LLM as representation model faster

### DISCUSSION THREAD

**Discussion by ASH1998** [MaartenGr/BERTopic#2086] [Q&A]
Hello, thanks for this amazing library. 
I am working on a project where I need to process millions of rows of data from storage using LLM (Azure OpenAI) as representation model. 

As far as i have seen the documentation, it seems like the calls happen sequentially, and it takes me days to complete the processing. 
Is there a way to make it faster using either: 

1. Concurrent Calls
2. Batching
3. Both

I have also removed umap with empty dimensionality model as it was also taking very long to process. 

Thanks!

**Comment by MaartenGr** [comment]
Thank you for the kind words!

I'm going to need a bit more information, such as the code that you have been using and the version of BERTopic. Also, what exactly is going slow for you? Is it the LLM or any other process? You can check this by setting `verbose=True` and sharing the output here.
